core/src/scene/sceneLoader.cpp

```diff
@@ -1049,6 +1024,69 @@ std::vector<StyleParam> SceneLoader::parseStyleParams(Node params, Scene& scene,
     return out;
 }
 
+StyleUniforms SceneLoader::parseStyleUniforms(const Node& value, Scene& scene) {
+    std::string type = "";
+    std::vector<UniformValue> uniformValues;
+    if (value.IsScalar()) { //float, int (bool), string (texture)
+        UniformValue uVal;
+        try {
+            uVal = value.as<float>();
+            type = "float";
+        } catch (const BadConversion& e) {
+            auto strVal = value.as<std::string>();
+            type = "int";
+            if (strVal == "true") {
+                uVal = 1;
+            } else if (strVal == "false") {
+                uVal = 0;
+            } else {
+                type = "sampler2D";
+                uVal = strVal;
+                auto texItr = scene.textures().find(strVal);
+                if (texItr == scene.textures().end()) {
+                    loadDefaultTexture(strVal, scene);
+                }
+            }
+        }
+        uniformValues.push_back(uVal);
+    } else if (value.IsSequence()) {
+        int size = value.size();
+        try {
+            type = "vec" + std::to_string(size);
+            UniformValue uVal;
+            switch (size) {
+                case 2:
+                    uVal = parseVec<glm::vec2>(value);
+                    break;
+                case 3:
+                    uVal = parseVec<glm::vec3>(value);
+                    break;
+                case 4:
+                    uVal = parseVec<glm::vec4>(value);
+                    break;
+                default:
+                    break;
+                    // TODO: Handle numeric arrays past 4 elements
```

Comment:
I do think it's worth doing this, and also that we can skip supporting `structs` for ES (and deprecate for JS).